### PR TITLE
Retain comments and linebreaks when writting new yaml

### LINKED ISSUE #17: Retain pubspec.yml contents
Hi there,

thanks for your awesome package! Is there a possibility to retain comments, linebreaks etc. when bumping the version? After bumping the version my pubspec file is very condensed and all comments and linebreaks are deleted.

### PR DESCRIPTION
**Pull request by Agraphie**
Resolves #17 by parsing and updating version in place.

### CODE REVIEW

**Comment by f3ath** [comment]
OK, so I discussed this with a friend. Yaml seems to be trickier than I though. For instance, what if the `version:` line has a comment in itself?

```yaml
version: 0.2.3 # lol parse me 
```

Or what if there is a multiline string?
```yaml
version: 0.2.3 # i am the version line
description: 'and I am a line with a
version: 3.2.1
in it'
```

I don't see an easy way to avoid using a real yaml parser.

**Comment by Agraphie** [comment]
Yes good thought, you are right about that. But nothing a little regex can't fix. We just use a positive lookahead and match the version which is not contained in `"` or `'`. Then we non-greedely match until `#` to exclude any comments behind the version. I also included the things you mentioned in the test yaml.

File: test/pubspec_sample.yaml [hunk 1]
```diff
@@ -1,6 +1,10 @@
-description: "A CLI tool to set/bump the `version` key in pubspec.yaml."
+description: "A CLI tool to set/bump the `version` key in pubspec.yaml.
+version: 1.2.3
```
Comment (f3ath) [inline test/pubspec_sample.yaml]: Is there a test checking this line is not getting changed?

File: lib/src/update_version.dart [hunk 2]
```diff
@@ -13,10 +14,28 @@ abstract class UpdateVersion extends Command {
...
+    Version newVersion = nextVersion(oldVersion);
+    console.log(newVersion.toString());
+
+    _writeNewPubSpec(pubSpec, oldVersion, newVersion);
+  }
+
+  Future _writeNewPubSpec(
+      File pubSpec, Version oldVersion, Version newVersion) async {
+    String newPubSpec = pubSpec.readAsStringSync().replaceFirst(
+        'version: "${oldVersion}"', "version: " + '"${newVersion.toString()}"');
```
Comment (f3ath) [inline lib/src/update_version.dart]: I don't understand how different quotation is processed. This line seems to only deal with double quotes. Is there a test for double/single/no quotes?

File: lib/src/version_extractor.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import 'dart:io';
+
+import 'package:pub_semver/pub_semver.dart';
+
+abstract class VersionExtractor {
```
Comment (f3ath) [inline lib/src/version_extractor.dart]: Why is it a class and why is the class abstract? A simple function would be enough.

**Review by f3ath — CHANGES_REQUESTED** [review]

**Pull request closed without merging**